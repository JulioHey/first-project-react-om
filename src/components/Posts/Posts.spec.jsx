import {
  render,
  screen
} from "@testing-library/react";

import {Posts} from '.';

const props = {
  posts: [
  {
    title: "title1",
    body: "body1",
    id: 1,
    cover: "img/img1.png",
  },
  {
    title: "title2",
    body: "body2",
    id: 2,
    cover: "img/img2.png",
  },
  {
    title: "title3",
    body: "body3",
    id: 3,
    cover: "img/img3.png",
  },
  {
    title: "title4",
    body: "body4",
    id: 4,
    cover: "img/img4.png",
  },
  ]
}

describe("<Posts />", () => {
  it ('should render posts', () => {
    render(<Posts {...props}/>);

    expect(screen.getAllByRole("heading", {name: /title/i}))
      .toHaveLength(4);
    expect(screen.getAllByRole("img", {name: /title/i}))
      .toHaveLength(4);
    expect(screen.getAllByText(/body/i))
      .toHaveLength(4);
    expect(screen.getByRole("img", {name: props.posts[0].title}))
      .toHaveAttribute("src", props.posts[0].cover);
  });

  it ('should not render if no posts', () => {
    render(<Posts />);
    expect(screen.queryByRole("heading", {name: /title/i}))
      .not.toBeInTheDocument();
  });

  it ('should match snapshot', () => {
    const {container} = render(<Posts {...props}/>);

    expect(container.firstChild).toMatchSnapshot();
  });
})