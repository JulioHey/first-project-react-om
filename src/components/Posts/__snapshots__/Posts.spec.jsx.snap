// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title1
      </h1>
      <p>
        body1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title2
      </h1>
      <p>
        body2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title3
      </h1>
      <p>
        body3
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title4"
      src="img/img4.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title4
      </h1>
      <p>
        body4
      </p>
    </div>
  </div>
</div>
`;
